controllers/operatorpolicy_status.go

```diff
@@ -1187,20 +1252,38 @@ var noExistingDeploymentObj = policyv1.RelatedObject{
 	Reason:    "No relevant deployments found",
 }
 
-// catalogSourceObj returns a conditionally compliant RelatedObject with reason based on the
-// `isUnhealthy` and `isMissing` parameters
-func catalogSourceObj(catalogName string, catalogNS string, isUnhealthy bool, isMissing bool) policyv1.RelatedObject {
+// catalogSourceObj returns a conditionally compliant RelatedObject with reason
+// based on the `isUnhealthy` and `isMissing` parameters. The `complianceConfig`
+// parameter determines whether an unhealthy CatalogSource should lead to
+// NonCompliance when status is updated.
+func catalogSourceObj(
+	catalogName string,
+	catalogNS string,
+	isUnhealthy bool,
+	isMissing bool,
+	complianceConfig policyv1beta1.ComplianceConfigAction,
+) policyv1.RelatedObject {
 	compliance := string(policyv1.Compliant)
 	reason := reasonWantFoundExists
 
 	if isUnhealthy {
-		compliance = string(policyv1.NonCompliant)
 		reason = reasonWantFoundExists + " but is unhealthy"
+
+		if complianceConfig != "Compliant" {
+			compliance = string(policyv1.NonCompliant)
+		} else {
+			reason += " (policy compliance is not impacted due to spec.complianceConfig.catalogSourceUnhealthy)"
```

Comment:
@mprahl how do you feel about this one. I think "___ exists but is unhealthy" on its own in a Compliant object could be confusing. It is the default behavior, but I feel like mentioning why "unhealthy" is ok in this case is good.